jbehave-support-core/src/main/java/org/jbehavesupport/core/internal/web/action/SelectWebAction.java

```diff
@@ -23,9 +23,17 @@ public void perform(WebActionContext ctx) {
 
         WebElement element = findElement(ctx);
 
-        Assert.isTrue(equalsIgnoreCase(element.getTagName(), "select"), "The SELECT action is not supported on this element type: " + element.getTagName());
-
-        new Select(element).selectByVisibleText(ctx.getData());
+        if (equalsIgnoreCase(element.getTagName(), "input") && equalsIgnoreCase(element.getAttribute("type"), "checkbox")){
+            boolean checked = ctx.getData().isEmpty() || toBoolean(ctx.getData());
+
+            if (toBoolean(element.getAttribute("checked")) ^ checked){
```

Comment:
minor: could you reformat the code please `){` > `)  {` (there should be `.idea\codeStyleSettings.xml` in current master branch)